### ISSUE TITLE
Crash around [FCMessageServices fetchAllChannels:]

### ISSUE THREAD

**Issue by graham-perks-snap** [freshdesk/freshchat-ios#24]
Yesterday we starting seeing several users experiencing crashes in Freshchat. 

We're unable to reproduce at will. I've emailed the full stack track to support@freshchat.com.

I'm just opening the issue here in case others are experiencing the same.

Partial stack:
```
__38+[FCMessageServices fetchAllChannels:]_block_invoke
__35-[FCAPIClient request:withHandler:]_block_invoke
__75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke
```

There's a bunch of CoreData & SQLite going on in a couple of background threads.

**Comment by graham-perks-snap** [comment]
Forgot to mention, we are on FreshchatSDK 1.5.5.

**Comment by graham-perks-snap** [comment]
Oddly while we had many crashes here yesterday, we have seen none today.

**Comment by NickAtLeafly** [comment]
Forgot to add that we received these crashes Dec 3rd. We've received no reports today of crashes.

**Comment by harish-kumarc** [comment]
Thanks for reporting the crash. We are investing it from our side and if you folks are able to replicate it please share the steps.
Also please let us know if background fetch is enabled and your app is using some location service(s)

Thanks

**Comment by harish-kumarc** [comment]
Thanks @NickAtLeafly for more information, we will check and update you. Also can you confirm that you are using native iOS SDK.

Thanks

**Comment by NickAtLeafly** [comment]
@harish-kumarc  Thank you very much. We are using the native iOS SDK.

**Comment by harish-kumarc** [comment]
Hi @NickAtLeafly , we have fixed the issue with version [v2.4.0](https://github.com/freshdesk/freshchat-ios/releases/tag/v2.4.0) . Please do let us know from your side also.

Thanks
